client/src/nonblocking/rpc_client.rs

```diff
@@ -0,0 +1,5065 @@
+/// creating `RpcClient` with an explicit [`CommitmentConfig`], and that default
+/// configured commitment level can be overridden by calling the various
+/// `_with_commitment` methods, like
+/// [`RpcClient::confirm_transaction_with_commitment`]. In some cases the
+/// configured commitment level is ignored and `Finalized` is used instead, as
+/// in [`RpcClient::get_blocks`], where it would be invalid to use the
+/// [`Processed`] commitment level. These exceptions are noted in the method
+/// documentation.
+///
+/// [`Finalized`]: CommitmentLevel::Finalized
+/// [`Processed`]: CommitmentLevel::Processed
+/// [jsonprot]: https://docs.solana.com/developing/clients/jsonrpc-api
+/// [JSON-RPC]: https://www.jsonrpc.org/specification
+/// [slots]: https://docs.solana.com/terminology#slot
+/// [cl]: https://docs.solana.com/developing/clients/jsonrpc-api#configuring-state-commitment
+///
+/// # Errors
+///
+/// Methods on `RpcClient` return
+/// [`client_error::Result`][crate::client_error::Result], and many of them
+/// return the [`RpcResult`][crate::rpc_response::RpcResult] typedef, which
+/// contains [`Response<T>`][crate::rpc_response::Response] on `Ok`. Both
+/// `client_error::Result` and [`RpcResult`] contain `ClientError` on error. In
+/// the case of `RpcResult`, the actual return value is in the
+/// [`value`][crate::rpc_response::Response::value] field, with RPC contextual
+/// information in the [`context`][crate::rpc_response::Response::context]
+/// field, so it is common for the value to be accessed with `?.value`, as in
+///
+/// ```
+/// # use solana_sdk::system_transaction;
+/// # use solana_client::rpc_client::RpcClient;
+/// # use solana_client::client_error::ClientError;
+/// # use solana_sdk::signature::{Keypair, Signer};
+/// # use solana_sdk::hash::Hash;
+/// # let rpc_client = RpcClient::new_mock("succeeds".to_string());
+/// # let key = Keypair::new();
+/// # let to = solana_sdk::pubkey::new_rand();
+/// # let lamports = 50;
+/// # let latest_blockhash = Hash::default();
+/// # let tx = system_transaction::transfer(&key, &to, lamports, latest_blockhash);
+/// let signature = rpc_client.send_transaction(&tx)?;
+/// let statuses = rpc_client.get_signature_statuses(&[signature])?.value;
+/// # Ok::<(), ClientError>(())
+/// ```
+///
+/// Requests may timeout, in which case they return a [`ClientError`] where the
+/// [`ClientErrorKind`] is [`ClientErrorKind::Reqwest`], and where the interior
+/// [`reqwest::Error`](crate::client_error::reqwest::Error)s
+/// [`is_timeout`](crate::client_error::reqwest::Error::is_timeout) method
+/// returns `true`. The default timeout is 30 seconds, and may be changed by
+/// calling an appropriate constructor with a `timeout` parameter.
+pub struct RpcClient {
+    sender: Box<dyn RpcSender + Send + Sync + 'static>,
+    config: RpcClientConfig,
+    node_version: RwLock<Option<semver::Version>>,
+}
+
+impl RpcClient {
```

Comment:
This is the original `RpcClient` implementation but `async`ed.

Reply:
I didn't review this closely. Assuming it's a strict copy + async/await tags added as needed.